Pathogens/Utility/location_utils/lat_lng_generation_report.py

```diff
@@ -0,0 +1,314 @@
+    text_as_string = mapbox_response.text if mapbox_response.text is not None else "N/A"
+    matching_text_as_string = mapbox_response.matching_text if mapbox_response.matching_text is not None else "N/A"
+    region_name_as_string = mapbox_response.region_name if mapbox_response.region_name is not None else "N/A"
+
+    return "[" + "center_coordinates: " + center_coordinates_as_string \
+        + ", bounding_box: " + bounding_box_as_string \
+        + ", text: " + text_as_string \
+        + ", matching_text: " + matching_text_as_string \
+        + ", region_name: " + region_name_as_string + "]"
+
+def generate_line_for_response_and_request(\
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    mapbox_request_url: str,\
+    mapbox_response\
+):
+    prefix = generate_mapbox_api_request_log_prefix(
+        logLevel = MapboxApiRequestLogLevel.INFO,
+        city_name = city_name,
+        state_name = state_name,
+        country_name = country_name
+    )
+    request_with_api_key_redacted = "[" + redact_mapbox_api_key_from_mapbox_api_query_url(mapbox_request_url) + "]"
+
+    return prefix + " - " + request_with_api_key_redacted + " - " + format_mapbox_response_as_string(mapbox_response)
+
+def generate_line_for_text_consitency_check(
+    city_name: str | None,\
+    state_name: str | None,\
+    country_name: str,\
+    mapbox_response,\
+):
+    if(mapbox_response is None):
+        return None
+    
+    expected_text = city_name if city_name is not None else state_name if state_name is not None else country_name
+
+    if(expected_text == mapbox_response.text):
+        return None
+
+    if(expected_text == mapbox_response.matching_text):
+        return None
+    
+    expected_text_to_display = '"' + expected_text + '"'
+    text_from_mapbox_api_to_display = format_mapbox_text_and_matching_text_for_display(mapbox_response)
+
+    prefix = generate_mapbox_api_request_log_prefix(
+        logLevel = MapboxApiRequestLogLevel.WARN,
+        city_name = city_name,
+        state_name = state_name,
+        country_name = country_name
+    )
+    info_statement = "[Text returned from the mapbox API did not match expected text]"
+    details = '[Expected text: ' + expected_text_to_display + ', Actual: ' + text_from_mapbox_api_to_display + ']'
+    suggestion = "[If one of the actual text values listed is the correct name of the city/state/country and no other errors or warnings are present, It might be appropriate to change the name of this location in Airtable to the actual name listed. Otherwise, please look to the other warnings and errors for this location for more guidance, the texts not matching might be indicative that mapbox has identified the wrong city based on the text it was given.]"
+
+    return prefix + " - " + info_statement + " - " + details + " - " + suggestion
+
+def generate_line_for_city_state_bounding_box_consistency_check(
```

Comment:
Context: This is a big one, basically looks for instances where the city listed doesn't belong to the state listed geographically (Ex. Something like Calgary, Saskatchewan, Canada). Root of a bunch of our wrong locations on the map. Should speak for itself (hopefully?) but hmu if this doesn't make sense.